src/Microsoft.ML.FastTree/FastTree.cs

```diff
@@ -3378,7 +3361,7 @@ public TreeNode(Dictionary<string, object> keyValues)
     /// and <see cref="TreeEnsembleModelParametersBasedOnRegressionTree"/> is the type of
     /// <see cref="TrainedTreeEnsemble"/>.
     /// </summary>
-    public abstract class TreeEnsembleModelParametersBasedOnRegressionTree : TreeEnsembleModelParameters
+    public abstract class TreeEnsembleModelParametersBasedOnRegressionTree : TreeEnsembleModelParameters, ICanGetSummaryAsIDataView
```

Comment:
Why do we need `ICanGetSummaryAs...`? `IDataView RegressionTreeEnsembleAsIDataView(..)` looks sufficient for generating a summary.

Reply:
That's the interface that Summarize entrypoint uses to get the IDataView from a trainer.

See: https://github.com/dotnet/machinelearning/blob/665a3669652bc478fffbc7ed73b89067e5a7938c/src/Microsoft.ML.Data/EntryPoints/SummarizePredictor.cs#L32

And:
https://github.com/dotnet/machinelearning/blob/665a3669652bc478fffbc7ed73b89067e5a7938c/src/Microsoft.ML.Data/EntryPoints/SummarizePredictor.cs#L49